desc/objectives/linear_objectives.py

```diff
@@ -39,6 +40,137 @@ def update_target(self, eq):
+    bounds : tuple of dict {float, ndarray}, optional
+        Lower and upper bounds on the objective. Overrides target.
+        Should have the same tree structure as thing.params.
+    weight : dict of {float, ndarray}, optional
+        Weighting to apply to the Objective, relative to other Objectives.
+        Should be a scalar or have the same tree structure as thing.params.
+    normalize : bool, optional
+        Whether to compute the error in physical units or non-dimensionalize.
+        Has no effect for this objective.
+    normalize_target : bool, optional
+        Whether target and bounds should be normalized before comparing to computed
+        values. If `normalize` is `True` and the target is in physical units,
+        this should also be set to True. Has no effect for this objective.
+    name : str, optional
+        Name of the objective function.
+
+    """
+
+    _scalar = False
+    _linear = True
+    _fixes = True
+    _units = "(~)"
+    _print_value_fmt = "Fixed parameter error: {:10.3e} "
+
+    def __init__(
+        self,
+        thing=None,
+        params=None,
+        indices=True,
+        target=None,
+        bounds=None,
+        weight=1,
+        normalize=False,
+        normalize_target=False,
+        name="Fixed parameter",
+    ):
+
+        self._target_from_user = target
+        self._params = params
+        self._indices = indices
+        super().__init__(
+            things=thing,
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            name=name,
+        )
+
+    def build(self, thing=None, use_jit=False, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        self.things = setdefault(thing, self.things)
+        thing = self.things[0]
+        params = setdefault(self._params, thing.optimizable_params)
+
+        if not isinstance(params, (list, tuple)):
+            params = [params]
+        assert all(par in thing.optimizable_params for par in params)
+        self._params = params
+
+        # replace indices=True with actual indices
+        if self._indices and isinstance(self._indices, bool):
+            self._indices = [np.arange(thing.dimensions[par]) for par in self._params]
+        # make sure its iterable if only a scalar was passed in
+        if not isinstance(self._indices, (list, tuple)):
+            self._indices = [self._indices]
+        # replace idx=True with array of all indices, throwing an error if the length
+        # of indices is different from number of params
+        indices = {}
+        for idx, par in zip(self._indices, self._params, strict=True):
```

Comment:
`strict` is only a keyword of `zip` in python 3.10 or later: https://peps.python.org/pep-0618/

so this would increase our minimum supported python version from 3.8 to 3.10